website/docs/r/security_group.html.markdown

```diff
@@ -25,30 +25,34 @@ Provides a security group resource.
 ```terraform
 resource "aws_security_group" "allow_tls" {
   name        = "allow_tls"
-  description = "Allow TLS inbound traffic"
+  description = "Allow TLS inbound traffic and all outbound traffic"
   vpc_id      = aws_vpc.main.id
 
-  ingress {
-    description      = "TLS from VPC"
-    from_port        = 443
-    to_port          = 443
-    protocol         = "tcp"
-    cidr_blocks      = [aws_vpc.main.cidr_block]
-    ipv6_cidr_blocks = [aws_vpc.main.ipv6_cidr_block]
-  }
-
-  egress {
-    from_port        = 0
-    to_port          = 0
-    protocol         = "-1"
-    cidr_blocks      = ["0.0.0.0/0"]
-    ipv6_cidr_blocks = ["::/0"]
-  }
-
   tags = {
     Name = "allow_tls"
   }
 }
+
+resource "aws_vpc_security_group_ingress_rule" "allow_tls" {
+  security_group_id = aws_security_group.allow_tls.id
+
+  cidr_ipv4   = [aws_vpc.main.cidr_block]
+  cidr_ipv6   = [aws_vpc.main.ipv6_cidr_block]
+  from_port   = 443
+  ip_protocol = "tcp"
+  to_port     = 443
+}
+
+resource "aws_vpc_security_group_egress_rule" "allow_all_traffic" {
+  security_group_id = aws_security_group.allow_tls.id
+
+  cidr_ipv4 = ["0.0.0.0/0"]
+  cidr_piv6 = ["::/0"]
+  # -1 is semantically equivalent to all ports and must be specified for the from_port and to_port if the protocol is "-1"
+  from_port   = -1
+  ip_protocol = "-1"
+  to_port     = -1
```

Comment:
tf accepts only one of the following [ cidr_ipv4  | cidr_ipv6 | prefix_list_id | referenced_security_group_id ] in a single rule definition.
Defining multiple destination types for a rule returns an error.

If ip_protocol = -1 and from_port = -1 and to_port = -1, it ends up with sg rule modification loop on each apply/plan
Since ip_protocol = -1 is translated at AWS's side to Protocol = All and Port range = All The proper code here would be:
```
resource "aws_vpc_security_group_egress_rule" "allow_all_traffic_ipv4" {
  security_group_id = aws_security_group.allow_tls.id
  cidr_ipv4 = ["0.0.0.0/0"]
  ip_protocol = "-1"
}

resource "aws_vpc_security_group_egress_rule" "allow_all_traffic_ipv6" {
  security_group_id = aws_security_group.allow_tls.id
  cidr_piv6 = ["::/0"]
  ip_protocol = "-1"
}
```

Reply:
Thanks for adding the proper code. I implemented my changes locally and it seemed to work so that's why I originally used it but I've changed it to your suggestion, thanks again for the help.